### ISSUE TITLE
some mistakes in bnn_layers.py

### ISSUE THREAD

**Issue by zhanghongjie101** [Jiaolong/xnor-net#1]
I think there are some mistakes in bnn_layers.py (mxnet version). In BinaryActivation, the forward and backward should contain the alpha. Do you think so? When I contain the alpha, I get 94.28% testing accuracy after 5 epochs.

**Comment by Jiaolong** [comment]
Thanks for your comments. This code is experimental and there were numerous bugs. I just fixed some of them. Now `BWN` should be around `97%` and `XNOR` around `96%`. 

Regarding to the `alpha` issue, it is indeed computed in the forward and backward of `BinaryConvolution` layer but not in the `BinaryActivation` layer. The scale factor of the binary inputs is not considered in this code, which  is the parameter `beta` in equation (10) of the XNOR-Net paper. If you included `beta`, it should produce better results.

**Comment by Jiaolong** [comment]
Just updated it again! With `Adam` optimizer and `lr=0.0002`, the results are better. 
```bash
python main.py --network=mnist_xnor
Namespace(batch_size=100, data_path='../data/MNIST_data/', learning_rate=0.1, momentum=0.9, network='mnist_xnor', num_epoch=5)
INFO:root:Epoch[0] Batch [200]	Speed: 2333.27 samples/sec	accuracy=0.893284
INFO:root:Epoch[0] Batch [400]	Speed: 2348.61 samples/sec	accuracy=0.973850
INFO:root:Epoch[0] Train-accuracy=0.976533
INFO:root:Epoch[0] Time cost=25.585
INFO:root:Epoch[0] Validation-accuracy=0.979300
INFO:root:Epoch[1] Batch [200]	Speed: 2370.62 samples/sec	accuracy=0.979751
INFO:root:Epoch[1] Batch [400]	Speed: 2381.95 samples/sec	accuracy=0.983550
INFO:root:Epoch[1] Train-accuracy=0.982764
INFO:root:Epoch[1] Time cost=25.247
INFO:root:Epoch[1] Validation-accuracy=0.980700
INFO:root:Epoch[2] Batch [200]	Speed: 2384.56 samples/sec	accuracy=0.982736
INFO:root:Epoch[2] Batch [400]	Speed: 2369.69 samples/sec	accuracy=0.985950
INFO:root:Epoch[2] Train-accuracy=0.984472
INFO:root:Epoch[2] Time cost=25.245
INFO:root:Epoch[2] Validation-accuracy=0.982100
INFO:root:Epoch[3] Batch [200]	Speed: 2362.13 samples/sec	accuracy=0.985920
INFO:root:Epoch[3] Batch [400]	Speed: 2351.47 samples/sec	accuracy=0.987700
INFO:root:Epoch[3] Train-accuracy=0.986834
INFO:root:Epoch[3] Time cost=25.455
INFO:root:Epoch[3] Validation-accuracy=0.980200
INFO:root:Epoch[4] Batch [200]	Speed: 2355.15 samples/sec	accuracy=0.986169
INFO:root:Epoch[4] Batch [400]	Speed: 2338.48 samples/sec	accuracy=0.987750
INFO:root:Epoch[4] Train-accuracy=0.988342
INFO:root:Epoch[4] Time cost=25.685
INFO:root:Epoch[4] Validation-accuracy=0.982600
Testing accuracy: 98.26%
```
